addons/multiplegap/src/presenter.js

```diff
@@ -1727,26 +1744,101 @@ function Addonmultiplegap_create(){
+                var studentAnswers = [];
+                studentAnswers = parseStudentAnswers (presenter.printableState);
+                if (showAnswers) {
+                    for (var i = 0; i < studentAnswers.length; i++){
+                        for (var j = 0; j < correctAnswers.length; j++){
+                            if(checkPrintedAnswer(studentAnswers[i], correctAnswers[j])){
+                                studentAnswers[i].isCorrect = true;
+                                break;
+                            }
+                        }
+                        answerHTML = prepareCheckedAnswerToPrint (answerHTML, i, ...studentAnswers);
+                    }
+                } else {
+                    answerHTML = prepareBasicPrint (answerHTML, ...studentAnswers);
+                }
             } else {
-                answerHTML = answerArray.join("</br>");
+                if (showAnswers) {
+                    answerHTML = prepareBasicPrint (answerHTML, ...correctAnswers);
+                }
             }
             $wrapper.html(answerHTML);
         }
 
         $view.append($wrapper);
 
-
         return $view[0].outerHTML;
     };
 
+    function prepareBasicPrint (answerHTML, ...answers) {
+        if (presenter.configuration.orientation === presenter.ORIENTATIONS.HORIZONTAL) {
+            for (var i = 0; i < answers.length; i++) {
+                answerHTML = answerHTML + answers[i].value;
+                if (i < answers.length - 1) {
+                    answerHTML = answerHTML + ", ";
+                }
+            }
+        } else {
+            for (var i = 0; i < answers.length; i++) {
+                answerHTML = answerHTML + answers[i].value;
+                if (i < answers.length - 1) {
+                    answerHTML = answerHTML + "</br>";
+                }
+            }
+        }
+        return answerHTML;
+    }
+
+    function parseStudentAnswers (state) {
+        var studentAnswers = [];
+        for (var i = 0; i < state.placeholders.length; i++){
+            var splitItemAnswerID = state.placeholders[i].item.split("-");
+            if (splitItemAnswerID.length === 2 && !isNaN(splitItemAnswerID[1])){
+                var answerAddonID = splitItemAnswerID[0];
+                var answerItemIndex = splitItemAnswerID[1] - 1;
+                var answer = {
+                    value : state.placeholders[i].value,
+                    addonID : answerAddonID,
+                    index : answerItemIndex,
+                    isCorrect : false
+                };
+                studentAnswers.push(answer);
+            }
+        }
+        return studentAnswers;
+    }
+
+    function prepareCheckedAnswerToPrint (answerHTML, i, ...answers){
+        if (answers[i].isCorrect) {
+            if (i < answers.length - 1) {
+                answers[i].value = '<div><span class=answerSpan>' + answers[i].value + '</span><span class="correctAnswerSpan"></span>, </div>';
```

Comment:
po pierwsze, generalnie lepiej  by było, żeby te elementy utworzyć "normalnie", tj. za pomocą document.createElement albo $('<div></div>'), poskładać z nich to co ci jest potrzebne, i kod html jako string wyciągać dopiero na samym końcu.